homeassistant/components/emulated_hue.py

```diff
@@ -77,7 +77,10 @@ def setup(hass, yaml_config):
         ssl_key=None,
         cors_origins=[],
         use_x_forwarded_for=False,
-        trusted_networks=[]
+        trusted_networks=[],
+        ip_bans=[],
+        login_threshold=0,
+        is_ban_enabled=False
```

Comment:
I realize that there are already mutable types in the kwargs in this function, but that in general is bad python and causes some gnarly bugs. Here is an example of why doing this is a very bad idea:

```bash
[jeff@omniscience ~]$ python3
Python 3.5.2 (default, Oct 14 2016, 12:54:53) 
[GCC 6.2.1 20160916 (Red Hat 6.2.1-2)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> def append_one(data=[]):
...     data.append(1)
...     return data
... 
>>> append_one()
[1]
>>> append_one()
[1, 1]
>>> append_one()
[1, 1, 1]
>>> append_one(data=[2])
[2, 1]
>>> append_one()
[1, 1, 1, 1]
>>> append_one()
[1, 1, 1, 1, 1]
>>> 
```

Since those variables such as `trusted_networks` or `ip_bans` are initialized (at module load time) with mutable types, the values stay as though they're globals along the life of the process. This means it stays the life of the entire hass daemon unless those kwargs are passed at function initialization time.

In theory setup() should only run once, but even still if possible can you not do this? I think setting them to `None` and then checking in the body of the function, while more verbose, will prevent an entire class of bugs related to object mutability in the future.